server/src/domain/asset/asset.service.ts

```diff
@@ -420,11 +436,82 @@ export class AssetService {
       await this.assetRepository.updateAll([options.stackParentId], { stackParentId: null });
     }
 
+    if (creationDate !== undefined || latitude !== undefined || longitude !== undefined) {
+      for (const id of ids) {
+        const { pathMetadata, newExifData } = await this.getPathAndExif(id, {
+          CreationDate: creationDate,
+          GPSLatitude: latitude,
+          GPSLongitude: longitude,
+        });
+        await this.assetRepository.save({ id, sidecarPath: pathMetadata });
+        await this.assetRepository.writeExif(pathMetadata, newExifData);
+        await this.jobRepository.queue({ name: JobName.METADATA_EXTRACTION, data: { id: id } });
+      }
+    }
+
     await this.jobRepository.queue({ name: JobName.SEARCH_INDEX_ASSET, data: { ids } });
     await this.assetRepository.updateAll(ids, options);
     this.communicationRepository.send(CommunicationEvent.ASSET_UPDATE, authUser.id, ids);
   }
 
+  async getPathAndExif(id: string, exifModify: Partial<Tags>): Promise<{ pathMetadata: string; newExifData: Tags }> {
+    const asset = await this.assetRepository.getById(id);
+    if (!asset) {
+      throw new BadRequestException('Asset not found');
+    }
+
+    let pathMetadata: string;
+    let exifData: Tags;
+
+    const nonWritableTags = [
+      'SourceFile',
+      'errors',
+      'ExifToolVersion',
+      'FileSize',
+      'FileAccessDate',
+      'FileInodeChangeDate',
+      'FileType',
+      'FileTypeExtension',
+      'MIMEType',
+      'tz',
+      'tzSource',
+    ];
```

Comment:
It might be more robust to operate with a list of `writableTags` instead?

Reply:
After a look at the code further down: Is this just a list of tags for which exiftool throws an error?

Reply:
Also fix with the exiftool append